### PR TITLE
Extract static pytest args into config

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Move all of the static pytest CLI args from the runtime scripts over into the config.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: setup.cfg
Comment (jezdez) [inline setup.cfg:1]: Let's use this opportunity to move this into the [`pyproject.toml`](https://docs.pytest.org/en/7.3.x/reference/customize.html#pyproject-toml)

Comment (jezdez) [reply setup.cfg:1]: (same for flake8)

**Review by jezdez — CHANGES_REQUESTED** [review]

File: setup.cfg
Comment (kenodegard) [reply setup.cfg:1]: `flake8` doesn't natively support `pyproject.toml` (see [docs](https://flake8.pycqa.org/en/latest/user/configuration.html#configuration-locations) and [original request response](https://github.com/PyCQA/flake8/issues/234#issuecomment-1206730688)), will need to use [`flake8-pyproject`](https://pypi.org/project/Flake8-pyproject/)

Comment (jezdez) [reply setup.cfg:1]: *eyeroll* that flake8 ticket, I remember seeing it before

Comment (jezdez) [reply setup.cfg:1]: What about resurrecting #12279?

Comment (jezdez) [reply setup.cfg:1]: (obviously out of scope of this PR)

Comment (kenodegard) [reply setup.cfg:1]: well for now I've combined it all using flake8-pyproject

happy to resurrect the ruff effort

Comment (jezdez) [reply setup.cfg:1]: that's okay, let's not waste the effort, we can revisit ruff separately, maybe discuss it in the team again

**Review by jezdez — APPROVED** [review]

**Comment by jezdez** [comment]
The test failures might be legit, any implicit config loading happening here?

**Review by jezdez — APPROVED** [review]

**Pull request merged**